packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -0,0 +1,258 @@
+
+    populateTransaction: {
+      ${values(contract.functions)
+        .map((v) => v[0])
+        .map(generatePopulateTransactionFunction)
+        .join('\n')}
+    };
+  }`
+
+  return template
+}
+
+export function codegenContractFactory(contract: Contract, abi: any, bytecode?: BytecodeWithLinkReferences): string {
+  const constructorArgs =
+    (contract.constructor && contract.constructor[0] ? generateInputTypes(contract.constructor[0].inputs) : '') +
+    `overrides?: ${contract.constructor[0]?.stateMutability === 'payable' ? 'PayableOverrides' : 'Overrides'}`
+  const constructorArgNamesWithoutOverrides =
+    contract.constructor && contract.constructor[0] ? generateParamNames(contract.constructor[0].inputs) : ''
+  const constructorArgNames = constructorArgNamesWithoutOverrides
+    ? `${constructorArgNamesWithoutOverrides}, overrides || {}`
+    : 'overrides || {}'
+  if (!bytecode) return codegenAbstractContractFactory(contract, abi)
+
+  // tsc with noUnusedLocals would complain about unused imports
+  const ethersImports: string[] = ['Signer']
+  if (constructorArgs.match(/\WBytesLike(\W|$)/)) ethersImports.push('BytesLike')
+  if (constructorArgs.match(/\WBigNumberish(\W|$)/)) ethersImports.push('BigNumberish')
+
+  const ethersContractImports: string[] = ['Contract', 'ContractFactory']
+  if (constructorArgs.match(/\WPayableOverrides(\W|$)/)) {
+    ethersContractImports.push('PayableOverrides')
+  } else {
+    ethersContractImports.push('Overrides')
+  }
+
+  return `
+  import { ${ethersImports.join(', ')} } from "ethers";
+  import { Provider, TransactionRequest } from '@ethersproject/providers';
+  import { ${ethersContractImports.join(', ')} } from "@ethersproject/contracts";
+
+  import { ${contract.name} } from "./${contract.name}";
+
+  export class ${contract.name}Factory extends ContractFactory {
+    ${generateFactoryConstructor(contract, bytecode)}
+    deploy(${constructorArgs}): Promise<${contract.name}> {
+      return super.deploy(${constructorArgNames}) as Promise<${contract.name}>;
+    }
+    getDeployTransaction(${constructorArgs}): TransactionRequest {
+      return super.getDeployTransaction(${constructorArgNames});
+    };
+    attach(address: string): ${contract.name} {
+      return super.attach(address) as ${contract.name};
+    }
+    connect(signer: Signer): ${contract.name}Factory {
+      return super.connect(signer) as ${contract.name}Factory;
+    }
+    static connect(address: string, signerOrProvider: Signer | Provider): ${contract.name} {
+      return new Contract(address, _abi, signerOrProvider) as ${contract.name};
+    }
+  }
+
+  const _abi = ${JSON.stringify(abi, null, 2)};
+
+  const _bytecode = "${bytecode.bytecode}";
+
+  ${generateLibraryAddressesInterface(contract, bytecode)}
+  `
+}
+
+export function codegenAbstractContractFactory(contract: Contract, abi: any): string {
+  return `
+  import { Contract, Signer } from "ethers";
+  import { Provider } from "ethers/providers";
```

Comment:
@krzkaczor is this ok, or it should be `import { Provider } from '@ethersproject/providers';` ?

Reply:
uhh good call. That's why we need tests + isolated example to catch such problems 😆

Reply:
Oh my bad.. fixing this for now. @krzkaczor I'll have a look at how other tests are written and add the same for v5 here.